File: backend/services/emailService.js
const nodemailer = require('nodemailer');

const transporter = nodemailer.createTransport({
  service: 'gmail',
  auth: {
    user: 'gangatharan949@gmail.com',
    pass: 'indb pgtm brux izqh'
  }
});

const getHtmlTemplate = (title, userName, contentHtml) => {
  return `
    <!DOCTYPE html>
    <html>
      <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>${title}</title>
        <style>
          body {
            font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif;
            background-color: #f3f4f6;
            margin: 0;
            padding: 0;
            -webkit-text-size-adjust: none;
            -ms-text-size-adjust: none;
          }
          .email-wrapper {
            width: 100%;
            background-color: #f3f4f6;
            padding: 20px 0;
          }
          .email-container {
            max-width: 580px;
            margin: 0 auto;
            background-color: #ffffff;
            border-radius: 16px;
            overflow: hidden;
            box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.05), 0 2px 4px -1px rgba(0, 0, 0, 0.03);
            border: 1px solid #e5e7eb;
          }
          .email-header {
            background: linear-gradient(135deg, #059669 0%, #047857 100%);
            padding: 30px 24px;
            text-align: center;
          }
          .email-header h1 {
            color: #ffffff;
            margin: 0;
            font-size: 26px;
            font-weight: 800;
            letter-spacing: 0.5px;
          }
          .email-body {
            padding: 32px 24px;
            color: #374151;
            line-height: 1.6;
          }
          .email-body h2 {
            font-size: 20px;
            color: #111827;
            margin-top: 0;
            margin-bottom: 16px;
          }
          .email-body p {
            margin-top: 0;
            margin-bottom: 16px;
            font-size: 15px;
          }
          .otp-code {
            display: inline-block;
            font-size: 32px;
            font-weight: 800;
            color: #047857;
            letter-spacing: 6px;
            padding: 12px 24px;
            background-color: #ecfdf5;
            border: 2px dashed #a7f3d0;
            border-radius: 12px;
            margin: 16px 0;
            text-align: center;
          }
          .btn-action {
            display: inline-block;
            background-color: #059669;
            color: #ffffff;
            text-decoration: none;
            padding: 12px 28px;
            font-weight: 600;
            border-radius: 10px;
            margin: 16px 0;
            text-align: center;
          }
          .info-table {
            width: 100%;
            border-collapse: collapse;
            margin: 16px 0;
          }
          .info-table td {
            padding: 10px 12px;
            border-bottom: 1px solid #f3f4f6;
            font-size: 14px;
          }
          .info-table td.label {
            font-weight: 600;
            color: #4b5563;
            width: 35%;
          }
          .info-table td.value {
            color: #1f2937;
          }
          .email-footer {
            background-color: #f9fafb;
            padding: 24px;
            text-align: center;
            border-top: 1px solid #f3f4f6;
            color: #6b7280;
            font-size: 12px;
          }
          .email-footer p {
            margin: 4px 0;
          }
          .email-footer a {
            color: #059669;
            text-decoration: none;
          }
        </style>
      </head>
      <body>
        <div class="email-wrapper">
          <div class="email-container">
            <div class="email-header">
              <h1>EcoSync</h1>
            </div>
            <div class="email-body">
              <h2>Hello ${userName || 'User'},</h2>
              ${contentHtml}
            </div>
            <div class="email-footer">
              <p>Together we can reduce electronic waste and build a sustainable future.</p>
              <p>© ${new Date().getFullYear()} EcoSync. All rights reserved.</p>
              <p>Need support? Contact us at <a href="mailto:support@ecosync.com">support@ecosync.com</a></p>
            </div>
          </div>
        </div>
      </body>
    </html>
  `;
};

// Helper function to send email via nodemailer
const sendMailHelper = async (to, subject, html) => {
  try {
    await transporter.sendMail({
      from: '"EcoSync" <gangatharan949@gmail.com>',
      to,
      subject,
      html
    });
    console.log(`[EMAIL SUCCESS] Sent "${subject}" successfully to ${to}`);
  } catch (error) {
    console.error(`[EMAIL ERROR] Failed to send "${subject}" to ${to}:`, error.message);
  }
};

/**
 * Sends a Registration OTP
 */
const sendOtp = async (email, otp, userName = 'User') => {
  const subject = 'EcoSync - Verify Your Email Address';
  const html = getHtmlTemplate(
    subject,
    userName,
    `<p>Thank you for signing up with EcoSync! To complete your registration, please verify your email address by entering the following 6-digit OTP code:</p>
     <div style="text-align: center;"><div class="otp-code">${otp}</div></div>
     <p>Please note that this code is valid for <strong>5 minutes</strong>. If you did not request this registration, you can safely ignore this email.</p>`
  );
  await sendMailHelper(email, subject, html);
};

/**
 * Sends Password Reset OTP
 */
const sendPasswordResetOtp = async (email, otp, userName = 'User') => {
  const subject = 'EcoSync - Password Reset OTP';
  const html = getHtmlTemplate(
    subject,
    userName,
    `<p>We received a request to reset your EcoSync password. Please enter the following 6-digit OTP code to authorize the password reset:</p>
     <div style="text-align: center;"><div class="otp-code">${otp}</div></div>
     <p>This code will expire in <strong>5 minutes</strong>. If you did not make this request, please ignore this email.</p>`
  );
  await sendMailHelper(email, subject, html);
};

/**
 * Sends a Welcome/Verification Success email
 */
const sendWelcomeEmail = async (email, userName = 'User') => {
  const subject = 'EcoSync - Welcome & Verification Success';
  const html = getHtmlTemplate(
    subject,
    userName,
    `<p>Your email address has been successfully verified! Welcome to the EcoSync family.</p>
     <p>Your account is now fully active. You can log in to schedule pickup requests, track your collections, and help recycle electronics responsibly.</p>
     <div style="text-align: center;"><a href="https://ewaste-frontend-4jyx.onrender.com/login" class="btn-action">Log In to Your Account</a></div>`
  );
  await sendMailHelper(email, subject, html);
};

/**
 * Sends a Request Submitted confirmation
 */
const sendPickupSubmitted = async (email, deviceType, quantity, address, userName = 'User') => {
  const subject = 'EcoSync - E-Waste Pickup Request Submitted';
  const html = getHtmlTemplate(
    subject,
    userName,
    `<p>We have successfully received your electronic waste pickup request. Here are the submission details:</p>
     <table class="info-table">
       <tr>
         <td class="label">Device Type:</td>
         <td class="value">${deviceType}</td>
       </tr>
       <tr>
         <td class="label">Quantity:</td>
         <td class="value">${quantity}</td>
       </tr>
       <tr>
         <td class="label">Pickup Address:</td>
         <td class="value">${address}</td>
       </tr>
       <tr>
         <td class="label">Status:</td>
         <td class="value" style="color: #d97706; font-weight: bold;">PENDING REVIEW</td>
       </tr>
     </table>
     <p>Our team is currently reviewing your submission. You will receive an email confirmation once the pickup is scheduled.</p>`
  );
  await sendMailHelper(email, subject, html);
};

/**
 * Sends Pickup Scheduled notification
 */
const sendPickupScheduled = async (email, deviceType, date, time, adminNotes, userName = 'User') => {
  const subject = 'EcoSync - E-Waste Pickup Scheduled';
  const html = getHtmlTemplate(
    subject,
    userName,
    `<p>Great news! Your electronic waste pickup request has been reviewed, approved, and scheduled for collection.</p>
     <table class="info-table">
       <tr>
         <td class="label">Device Type:</td>
         <td class="value">${deviceType}</td>
       </tr>
       <tr>
         <td class="label">Scheduled Date:</td>
         <td class="value" style="font-weight: bold; color: #047857;">${date}</td>
       </tr>
       <tr>
         <td class="label">Scheduled Time:</td>
         <td class="value" style="font-weight: bold; color: #047857;">${time}</td>
       </tr>
       ${adminNotes ? `<tr><td class="label">Admin Notes:</td><td class="value">${adminNotes}</td></tr>` : ''}
       <tr>
         <td class="label">Status:</td>
         <td class="value" style="color: #059669; font-weight: bold;">SCHEDULED</td>
       </tr>
     </table>
     <p>Please keep the devices packed and ready by the scheduled slot. Thank you for your contribution to a greener environment!</p>`
  );
  await sendMailHelper(email, subject, html);
};

/**
 * Sends Better Images Required notification
 */
const sendBetterImagesRequired = async (email, deviceType, userName = 'User') => {
  const subject = 'EcoSync - Action Required: Upload Better Images';
  const html = getHtmlTemplate(
    subject,
    userName,
    `<p>Our collection team inspected the details of your e-waste request for: <strong>${deviceType}</strong>.</p>
     <p>To safely approve and plan the logistics, we need clearer images of the electronic item. Please log in to your dashboard and update the images for this pickup request.</p>
     <div style="text-align: center;"><a href="https://ewaste-frontend-4jyx.onrender.com/user/my-requests" class="btn-action">Update Request Details</a></div>`
  );
  await sendMailHelper(email, subject, html);
};

/**
 * Sends status updates (Completed, Rejected, etc.)
 */
const sendTrackingStatusUpdate = async (email, deviceType, requestId, status, adminNotes = '', userName = 'User') => {
  let subject = `EcoSync - Pickup Request Update (#${requestId})`;
  let content = '';

  if (status === 'REJECTED') {
    subject = 'EcoSync - Pickup Request Rejected';
    content = `<p>We regret to inform you that your e-waste pickup request has been rejected after review.</p>
               <table class="info-table">
                 <tr>
                   <td class="label">Device:</td>
                   <td class="value">${deviceType}</td>
                 </tr>
                 <tr>
                   <td class="label">Status:</td>
                   <td class="value" style="color: #dc2626; font-weight: bold;">REJECTED</td>
                 </tr>
                 ${adminNotes ? `<tr><td class="label">Reason/Notes:</td><td class="value">${adminNotes}</td></tr>` : ''}
               </table>
               <p>If you believe this was an error, please submit a new request with correct details via your dashboard.</p>`;
  } else if (status === 'COMPLETED' || status === 'COLLECTED') {
    subject = 'EcoSync - E-Waste Pickup Completed';
    content = `<p>Success! Your electronic waste collection has been successfully completed by our team.</p>
               <table class="info-table">
                 <tr>
                   <td class="label">Device:</td>
                   <td class="value">${deviceType}</td>
                 </tr>
                 <tr>
                   <td class="label">Status:</td>
                   <td class="value" style="color: #059669; font-weight: bold;">COMPLETED / RECYCLED</td>
                 </tr>
               </table>
               <p>The collected devices have been securely routed to our recycling facilities. Your recycling effort has successfully prevented hazardous chemicals from reaching landfills. Thank you!</p>`;
  } else {
    content = `<p>Your pickup request status has been updated to: <strong>${status}</strong>.</p>
               <p>Log in to your dashboard to track details or contact support for queries.</p>`;
  }

  const html = getHtmlTemplate(subject, userName, content);
  await sendMailHelper(email, subject, html);
};

module.exports = {
  sendOtp,
  sendWelcomeEmail,
  sendPasswordResetOtp,
  sendPickupSubmitted,
  sendPickupScheduled,
  sendBetterImagesRequired,
  sendTrackingStatusUpdate
};

File: backend/src/main/java/com/ewaste/email/EmailService.java
package com.ewaste.email;

import com.ewaste.model.EwasteRequest;
import com.ewaste.model.User;
import jakarta.mail.MessagingException;
import jakarta.mail.internet.MimeMessage;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;

import java.time.format.DateTimeFormatter;

@Service
@RequiredArgsConstructor
@Slf4j
public class EmailService {

    private final JavaMailSender mailSender;

    @Value("${app.email.from}")
    private String fromEmail;

    // ─── Registration OTP ──────────────────────────────────────────────────
    @Async
    public void sendRegistrationOtp(String email, String otp) {
        try {
            String html = buildOtpEmail(
                "Verify Your Email", email, otp,
                "Use this OTP to complete your EcoCollect registration.",
                "This OTP expires in <strong>5 minutes</strong>."
            );
            sendHtmlEmail(email, "EcoCollect – Email Verification OTP", html);
            log.info("Registration OTP sent to {}", email);
        } catch (Exception e) {
            log.error("Failed to send registration OTP to {}: {}", email, e.getMessage());
        }
    }

    // ─── Forgot Password OTP ───────────────────────────────────────────────
    @Async
    public void sendForgotPasswordOtp(String email, String otp) {
        try {
            String html = buildOtpEmail(
                "Reset Your Password", email, otp,
                "Use this OTP to reset your EcoCollect password.",
                "This OTP expires in <strong>5 minutes</strong>. If you did not request this, ignore this email."
            );
            sendHtmlEmail(email, "EcoCollect – Password Reset OTP", html);
            log.info("Password reset OTP sent to {}", email);
        } catch (Exception e) {
            log.error("Failed to send password reset OTP: {}", e.getMessage());
        }
    }

    // ─── Request Submit OTP ────────────────────────────────────────────────
    @Async
    public void sendRequestSubmitOtp(String email, String name, String otp, String deviceType) {
        try {
            String html = buildOtpEmail(
                "Verify Your e-Waste Request", email, otp,
                "You're submitting a pickup request for <strong>" + deviceType + "</strong>. Use this OTP to verify.",
                "This OTP expires in <strong>5 minutes</strong>. The request will not be processed without verification."
            );
            sendHtmlEmail(email, "EcoCollect – Verify Your e-Waste Request", html);
            log.info("Request submit OTP sent to {}", email);
        } catch (Exception e) {
            log.error("Failed to send request OTP: {}", e.getMessage());
        }
    }

    // ─── Welcome Email ─────────────────────────────────────────────────────
    public void sendWelcomeEmail(User user) {
        try {
            String html = buildWelcomeEmail(user.getFirstName());
            sendHtmlEmail(user.getEmail(), "Welcome to EcoCollect! 🌿", html);
        } catch (Exception e) {
            log.error("Failed to send welcome email: {}", e.getMessage());
        }
    }

    // ─── Request Confirmation ──────────────────────────────────────────────
    public void sendRequestConfirmation(EwasteRequest req) {
        try {
            sendHtmlEmail(
                req.getUser().getEmail(),
                "Request Confirmed – #" + req.getId() + " | EcoCollect",
                buildRequestConfirmationEmail(req)
            );
        } catch (Exception e) {
            log.error("Failed to send confirmation: {}", e.getMessage());
        }
    }

    // ─── Pickup Scheduled ─────────────────────────────────────────────────
    public void sendPickupScheduledEmail(EwasteRequest req) {
        try {
            sendHtmlEmail(
                req.getUser().getEmail(),
                "Pickup Scheduled – " + formatDate(req) + " | EcoCollect",
                buildScheduleEmail(req)
            );
        } catch (Exception e) {
            log.error("Failed to send schedule email: {}", e.getMessage());
        }
    }

    // ─── Rejection ────────────────────────────────────────────────────────
    public void sendRejectionEmail(EwasteRequest req) {
        try {
            sendHtmlEmail(req.getUser().getEmail(),
                "Request Update – #" + req.getId() + " | EcoCollect",
                buildRejectionEmail(req));
        } catch (Exception e) {
            log.error("Failed to send rejection email: {}", e.getMessage());
        }
    }

    // ─── Completion ───────────────────────────────────────────────────────
    public void sendCompletionEmail(EwasteRequest req) {
        try {
            sendHtmlEmail(req.getUser().getEmail(),
                "Pickup Completed – Thank You! | EcoCollect",
                buildCompletionEmail(req));
        } catch (Exception e) {
            log.error("Failed to send completion email: {}", e.getMessage());
        }
    }

    // ─── Core send ────────────────────────────────────────────────────────
    private void sendHtmlEmail(String to, String subject, String html) throws MessagingException {
        MimeMessage message = mailSender.createMimeMessage();
        MimeMessageHelper helper = new MimeMessageHelper(message, true, "UTF-8");
        helper.setFrom(fromEmail);
        helper.setTo(to);
        helper.setSubject(subject);
        helper.setText(html, true);
        mailSender.send(message);
    }

    private String formatDate(EwasteRequest req) {
        if (req.getScheduledDate() == null) return "";
        return req.getScheduledDate().format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
    }

    // ─── HTML Templates ───────────────────────────────────────────────────

    private String wrapper(String content) {
        return """
        <!DOCTYPE html><html><head><meta charset="UTF-8">
        <meta name="viewport" content="width=device-width,initial-scale=1">
        <style>
          *{margin:0;padding:0;box-sizing:border-box}
          body{background:#0f172a;font-family:'Segoe UI',Arial,sans-serif;color:#e2e8f0}
          .wrap{max-width:560px;margin:32px auto;padding:0 16px}
          .header{background:linear-gradient(135deg,#0f172a 0%%,#1e3a2f 100%%);border:1px solid #22c55e33;border-radius:16px 16px 0 0;padding:28px 32px;display:flex;align-items:center;gap:12px}
          .logo{width:40px;height:40px;background:rgba(34,197,94,0.15);border:1px solid rgba(34,197,94,0.4);border-radius:12px;display:flex;align-items:center;justify-content:center;font-size:20px}
          .logo-text{font-size:22px;font-weight:800;color:#f8fafc;letter-spacing:-0.02em}
          .body{background:#1e293b;border:1px solid #334155;border-top:none;padding:32px}
          .footer{background:#0f172a;border:1px solid #334155;border-top:none;border-radius:0 0 16px 16px;padding:18px 32px;text-align:center;font-size:12px;color:#475569}
          h1{font-size:22px;font-weight:700;color:#f1f5f9;margin-bottom:10px}
          p{font-size:14px;color:#94a3b8;line-height:1.7;margin-bottom:14px}
          .otp-box{background:#0f172a;border:2px dashed #22c55e66;border-radius:14px;padding:28px;text-align:center;margin:24px 0}
          .otp-code{font-size:46px;font-weight:800;letter-spacing:12px;color:#22c55e;font-family:monospace}
          .otp-label{font-size:12px;color:#64748b;margin-top:8px;text-transform:uppercase;letter-spacing:.08em}
          .info-box{background:#0f172a;border:1px solid #334155;border-radius:10px;padding:18px;margin:18px 0}
          .info-row{display:flex;justify-content:space-between;padding:7px 0;border-bottom:1px solid #1e293b;font-size:13px}
          .info-row:last-child{border-bottom:none}
          .il{color:#64748b}.iv{color:#e2e8f0;font-weight:600}
          .highlight{color:#22c55e;font-weight:600}
          .badge{display:inline-block;padding:3px 12px;border-radius:100px;font-size:11px;font-weight:700;text-transform:uppercase;letter-spacing:.06em}
          .b-sched{background:rgba(96,165,250,.15);color:#60a5fa;border:1px solid rgba(96,165,250,.3)}
          .b-done{background:rgba(34,197,94,.15);color:#22c55e;border:1px solid rgba(34,197,94,.3)}
          .b-rej{background:rgba(248,113,113,.15);color:#f87171;border:1px solid rgba(248,113,113,.3)}
          .warn{background:rgba(251,191,36,.08);border:1px solid rgba(251,191,36,.25);border-radius:8px;padding:10px 14px;color:#fbbf24;font-size:13px;margin-top:8px}
        </style></head><body><div class="wrap">
        <div class="header"><div class="logo">♻</div><span class="logo-text">EcoCollect</span></div>
        <div class="body">%s</div>
        <div class="footer">© 2025 EcoCollect · Smart e-Waste Management<br>This is an automated email — please do not reply.</div>
        </div></body></html>
        """.formatted(content);
    }

    private String buildOtpEmail(String title, String email, String otp, String context, String expiry) {
        return wrapper("""
            <h1>%s</h1>
            <p>%s</p>
            <div class="otp-box">
              <div class="otp-code">%s</div>
              <div class="otp-label">Your One-Time Password</div>
            </div>
            <p style="text-align:center;font-size:13px;">⏱ %s</p>
            <div class="warn">🔒 Never share this OTP with anyone. EcoCollect will never ask for your OTP.</div>
            """.formatted(title, context, otp, expiry));
    }

    private String buildWelcomeEmail(String firstName) {
        return wrapper("""
            <h1>Welcome to EcoCollect, %s! 🌿</h1>
            <p>Your account has been verified and created successfully. You're now part of a community making e-waste disposal responsible and easy.</p>
            <div class="info-box">
              <div class="info-row"><span class="il">Submit Requests</span><span class="iv">Log e-waste for pickup</span></div>
              <div class="info-row"><span class="il">Track Status</span><span class="iv">Real-time updates</span></div>
              <div class="info-row"><span class="il">Get Notified</span><span class="iv">Email at every step</span></div>
              <div class="info-row"><span class="il">Map Pickup</span><span class="iv">Pin your exact location</span></div>
            </div>
            <p>Start by submitting your first e-waste pickup request today. Together, let's build a greener planet. 🌍</p>
            """.formatted(firstName));
    }

    private String buildRequestConfirmationEmail(EwasteRequest req) {
        return wrapper("""
            <h1>Request Verified & Submitted! ✅</h1>
            <p>Hi <span class="highlight">%s</span>, your e-waste pickup request has been verified and submitted for review.</p>
            <div class="info-box">
              <div class="info-row"><span class="il">Request ID</span><span class="iv">#%d</span></div>
              <div class="info-row"><span class="il">Device</span><span class="iv">%s – %s %s</span></div>
              <div class="info-row"><span class="il">Condition</span><span class="iv">%s</span></div>
              <div class="info-row"><span class="il">Quantity</span><span class="iv">%d</span></div>
              <div class="info-row"><span class="il">Pickup Address</span><span class="iv">%s</span></div>
            </div>
            <p>Our admin team will review your request and schedule a pickup shortly.</p>
            """.formatted(
                req.getUser().getFirstName(), req.getId(),
                req.getDeviceType(), req.getBrand(), req.getModel(),
                req.getCondition(), req.getQuantity(), req.getPickupAddress()));
    }

    private String buildScheduleEmail(EwasteRequest req) {
        String notes = (req.getAdminNotes() != null && !req.getAdminNotes().isBlank())
            ? "<p><strong>Note from our team:</strong> " + req.getAdminNotes() + "</p>" : "";
        return wrapper("""
            <h1>Pickup Scheduled! 📅</h1>
            <p>Hi <span class="highlight">%s</span>, your e-waste pickup has been confirmed.</p>
            <div class="info-box">
              <div class="info-row"><span class="il">Request ID</span><span class="iv">#%d</span></div>
              <div class="info-row"><span class="il">Device</span><span class="iv">%s – %s %s</span></div>
              <div class="info-row"><span class="il">Pickup Date</span><span class="iv" style="color:#60a5fa">%s</span></div>
              <div class="info-row"><span class="il">Pickup Time</span><span class="iv" style="color:#60a5fa">%s</span></div>
              <div class="info-row"><span class="il">Address</span><span class="iv">%s</span></div>
              <div class="info-row"><span class="il">Status</span><span class="iv"><span class="badge b-sched">SCHEDULED</span></span></div>
            </div>
            %s
            <p>Please ensure your e-waste is accessible on the scheduled day. Thank you for recycling responsibly! 🌿</p>
            """.formatted(
                req.getUser().getFirstName(), req.getId(),
                req.getDeviceType(), req.getBrand(), req.getModel(),
                formatDate(req),
                req.getScheduledTime() != null ? req.getScheduledTime() : "TBD",
                req.getPickupAddress(), notes));
    }

    private String buildRejectionEmail(EwasteRequest req) {
        return wrapper("""
            <h1>Request Update</h1>
            <p>Hi <span class="highlight">%s</span>, your pickup request <strong>#%d</strong> could not be processed at this time.</p>
            <div class="info-box">
              <div class="info-row"><span class="il">Request ID</span><span class="iv">#%d</span></div>
              <div class="info-row"><span class="il">Device</span><span class="iv">%s – %s %s</span></div>
              <div class="info-row"><span class="il">Status</span><span class="iv"><span class="badge b-rej">REJECTED</span></span></div>
            </div>
            <p>Please feel free to submit a new request or contact our support team.</p>
            """.formatted(
                req.getUser().getFirstName(), req.getId(), req.getId(),
                req.getDeviceType(), req.getBrand(), req.getModel()));
    }

    private String buildCompletionEmail(EwasteRequest req) {
        return wrapper("""
            <h1>Pickup Completed! 🎉</h1>
            <p>Hi <span class="highlight">%s</span>, your e-waste pickup <strong>#%d</strong> is done!</p>
            <div class="info-box">
              <div class="info-row"><span class="il">Request ID</span><span class="iv">#%d</span></div>
              <div class="info-row"><span class="il">Device</span><span class="iv">%s – %s %s</span></div>
              <div class="info-row"><span class="il">Status</span><span class="iv"><span class="badge b-done">COMPLETED</span></span></div>
            </div>
            <p>Your e-waste has been collected and will be recycled by our certified partners. Thank you for contributing to a greener planet! 🌍</p>
            """.formatted(
                req.getUser().getFirstName(), req.getId(), req.getId(),
                req.getDeviceType(), req.getBrand(), req.getModel()));
    }
}
